src/devices/README.md

```diff
@@ -251,6 +255,138 @@ iot/
+| ```public void SetEnvironmentData(Temperature temperature, Ratio humidity)``` |
+| In this example, the function Set the environement values of temperature and humidity. It makes it very easy to understand, on top of the XML comment what the function is doing. |
+
+| Do |
+|---|
+| Use properties for a single element Set and/or Get function |
+| ```public OperationMode OperationMode``` |
+| This function then can be implemented with a get and set: |
+
+```csharp
+public OperationMode OperationMode
+{
+    get
+    {
+        var mode = ReadRegister(Register.MEAS_MODE);
+        mode = (byte)((mode >> 4) & 0b0000_0111);
+        return (OperationMode)mode;
+    }
+
+    set
+    {
+        var mode = ReadRegister(Register.MEAS_MODE);
+        // Clear previous mode
+        mode = (byte)(mode & 0b1000_1111);
+        mode = (byte)(mode | (((byte)(value)) << 4));
+        WriteRegister(Register.MEAS_MODE, mode);
+    }
+}
+```
+
+| Do | Don't |
+|---|---|
+| Use full name like ```EquivalentTotalVolatileOrganicCompound``` | Don't use the abbreviation ```eTVOC``` |
+| This makes it easy for non specialist to use a binding. | If you are not a specialist or depending of the context, the abbreviation is complicated to understand. |
+
+### Commenting the code
+
+| Do |
+|---|
+| Use the ```<remarks>``` from the XML comments to add any important element regarding a function. |
+
+```csharp
+/// <remarks>A previously stored value may be written back to this two byte
+/// register and the Algorithms will use the new value in its
+/// calculations(until it adjusts it as part of its internal Automatic
+/// Baseline Correction). Please refer to documentation to understand when to restore a
+/// previous baseline: https://www.sciosense.com/wp-content/uploads/2020/01/Application-Note-Baseline-Save-and-Restore-on-CCS811.pdf</remarks>
+
+```
+
+| Do |
+|---|
+| Always reference specific elements with a poiunter documentation/reference to a specific part of the docuementation. This does including specific timing wait, intiailization phase and anything that would trigger a question like ```why this wait?``` or ```Why this function has to be called 2 times in a row```. If the documentation is the same for all the sensor implementation, refer the documentation once at the beginning and then mention the page. |
+| For example: |
+
+```csharp
+_controller.Write(_pinWake, PinValue.Low);
+// Doc says wait 50 micro seconds, page 58 https://urltothedoc
+DelayHelper.DelayMicroseconds(50, true);
+```
+
+### GpioController
+
+| Do|
+|---|
+| In a constructor if you are using a ```GpioController``` make sure you are using as well a bollean ```_shouldDispose```. Some platforms or some specific doungles like FT4222 does allow the creation of one controller so if used with multiple bindings, it can't be disposed just like that. In the ```Dispose``` function, make sure you close the possible open pins if ```_shouldDispose``` is false. |
```

Comment:
```suggestion
| In a constructor if you are using a ```GpioController``` make sure you are using as well a boolean ```_shouldDispose```. Some platforms or some specific doungles like FT4222 does allow the creation of one controller so if used with multiple bindings, it can't be disposed just like that. In the ```Dispose``` function, make sure you close the possible open pins if ```_shouldDispose``` is false. |
```